### PR TITLE
[DNM] store/tikv: use grpc client to interact with tikv

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR tries to rewrite the previous PR #3043. It focuses on making change minimal so that it is not scary for review.

It contains

1. The proto update, which contains the changes from a "make update" to update kvproto in `_vendor`
2. The interface changes, which includes changes for `Conn`, `Client`, `RegionRequestSender`, `tikvStore`, etc. to adapt the proto update at 1.
3. Changes to cover all callings of changed interfaces in 2.
4. Changes to ensure all test cases related to 1., 2., 3 pass

Since grpc use frames to multiplex a single tcp connection for requests, `rpcClient` could reuse a `grpc.ClientConn` for multiple requests. This refactor for connection pool will be left for a PR in future.

### CODE REVIEW

**Comment by hhkbp2** [comment]
PTAL @coocood @tiancaiamao @disksing @AndreMouche @siddontang @zimulala

File: store/tikv/client.go [hunk 1]
```diff
@@ -15,135 +15,178 @@
...
+	rpcLabelKV  = "kv"
+	rpcLabelCop = "cop"
 )
 
+// Client is a client that sends RPC.
+// It should not be used after calling Close().
+type Client interface {
+	// Close should release all data.
+	Close() error
+	// SendKVReq sends kv request.
```
Comment (tiancaiamao) [inline store/tikv/client.go]: Update comment

File: store/tikv/mock-tikv/rpc.go [hunk 2]
```diff
@@ -14,88 +14,93 @@
...
-	case kvrpcpb.MessageType_CmdRawDelete:
-		resp.CmdRawDeleteResp = h.onRawDelete(req.CmdRawDeleteReq)
-	}
-	resp.Type = req.Type
-	return &resp
-}
-
-func (h *rpcHandler) checkContext(ctx *kvrpcpb.Context) *errorpb.Error {
+func (h *rpcHandler) checkRequestContext(ctx *kvrpcpb.Context) *errorpb.Error {
 	ctxPear := ctx.GetPeer()
```
Comment (tiancaiamao) [inline store/tikv/mock-tikv/rpc.go]: s/ctxPear/ctxPeer

File: store/tikv/mock-tikv/rpc.go [hunk 3]
```diff
@@ -537,3 +418,179 @@ func NewRPCClient(cluster *Cluster, mvccStore *MvccStore) *RPCClient {
...
+			}
+
+			executors := extractExecutors(sel)
+			dag := &tipb.DAGRequest{
+				StartTs:        sel.GetStartTs(),
+				TimeZoneOffset: sel.TimeZoneOffset,
+				Flags:          sel.Flags,
+				Executors:      executors,
+			}
+			r.Data, err = dag.Marshal()
```
Comment (tiancaiamao) [inline store/tikv/mock-tikv/rpc.go]: r.Data = r.Data[:0]  // reset the Data before reuse it?
By the way, why Cop request proto is different from others ?

Comment (hhkbp2) [inline store/tikv/mock-tikv/rpc.go]: It's unnecessary to reset `Data`. The handling of Cop request is copied literally from the old code, which converts the request to DAG format. So it looks different.

**Comment by hhkbp2** [comment]
This PR is Replaced by #3138

**Pull request closed without merging**